libbeat/monitoring/report/elasticsearch/client.go

```diff
@@ -113,22 +115,15 @@ func (c *publishClient) Publish(batch publisher.Batch) error {
 			}
 		}
 
-		action := common.MapStr{
-			"index": common.MapStr{
-				"_type":    t,
-				"_index":   "",
-				"_routing": nil,
-			},
-		}
-		document := report.Event{
-			Timestamp: event.Content.Timestamp,
-			Fields:    event.Content.Fields,
+		switch params["format"] {
+		case "production":
+			err = c.bulkToProduction(params, event, t)
+		case "monitoring":
+			err = c.bulkToMonitoring(event)
+		default:
+			err = fmt.Errorf("unsupported reporting format: %v", params["format"])
 		}
```

Comment:
I always feel a little uneasy about comparison on constant strings spread out in the code. As the format doesn't change I'd prefer to have an enum or bool flag which must be set when calling the constructor. 

The error message should be returned on init, and not when publishing is ongoing.  => Better have constructor check for it.